pkg/op/token.go

```diff
@@ -102,7 +104,20 @@ func CreateJWT(ctx context.Context, issuer string, tokenRequest TokenRequest, ex
 	claims := oidc.NewAccessTokenClaims(issuer, tokenRequest.GetSubject(), tokenRequest.GetAudience(), exp, id, client.GetID(), client.ClockSkew())
 	if client != nil {
 		restrictedScopes := client.RestrictAdditionalAccessTokenScopes()(tokenRequest.GetScopes())
-		privateClaims, err := storage.GetPrivateClaimsFromScopes(ctx, tokenRequest.GetSubject(), client.GetID(), removeUserinfoScopes(restrictedScopes))
+
+		var (
+			privateClaims map[string]interface{}
+			err           error
+		)
+		if tokenExchangeRequest, ok := tokenRequest.(TokenExchangeRequest); ok {
+			privateClaims, err = storage.(TokenExchangeStorage).GetPrivateClaimsFromTokenExchangeRequest(
```

Comment:
Don't we need to check the assertion here to avoid a panic and return an error if the assertion failed?

Reply:
We do need to check after both assertions. Here a suggestion:

```suggestion
		tokenExchangeRequest, okReq := tokenRequest.(TokenExchangeRequest)
		teStorage, okStorage := storage.(TokenExchangeStorage)
		if okReq && okStorage {
			privateClaims, err = teStorage.GetPrivateClaimsFromTokenExchangeRequest(
``````suggestion
		teStorage, okStorage := storage.(TokenExchangeStorage)
``````suggestion
		teStorage, okStorage := storage.(TokenExchangeStorage)
``````suggestion
		teStorage, okStorage := storage.(TokenExchangeStorage)
``````suggestion
		teStorage, okStorage := storage.(TokenExchangeStorage)
``````suggestion
		teStorage, okStorage := storage.(TokenExchangeStorage)
```